pkgs/development/compilers/llvm/common/default.nix

```diff
@@ -1073,7 +1086,12 @@ let
           url = "https://github.com/llvm/llvm-project/pull/99837/commits/14ae0a660a38e1feb151928a14f35ff0f4487351.patch";
           hash = "sha256-JykABCaNNhYhZQxCvKiBn54DZ5ZguksgCHnpdwWF2no=";
           relative = "compiler-rt";
-        });
+        })
+        # Fixes baremetal
```

Comment:
This comment could do with refinement; it's not fixing baremetal (baremetal is fine), it's fixing the linux libc-free build.

Upstream appears to indicate a more appropriate fix would be to provide libc headers to the compiler-rt builtins build. I don't know if that's something tractable or not in the various possible useLLVM scenarios. Could use some help from others interested in determining whether/how this might impact other platforms such as Darwin. cc @emilazy when you're back and @Ericson2314.

Reply:
Another issue here:

```
nix eval nixpkgs/master#llvmPackages_git.compiler-rt
«error: path '/nix/store/54xn52z7jbib8ryvz6bw244ia34g5hw3-source/pkgs/development/compilers/llvm/git/compiler-rt/libc-free.patch' does not exist»
```

Reply:
I've opened up #380213 to take care of things